### ISSUE TITLE
Streamz aggregations fail with TypeError

### ISSUE THREAD

**Issue by inc0** [python-streamz/streamz#430]
Hello, I'm trying to write custom aggegator for streams. Streamz dataframe looks exactly what I need, but I'm having issues with time windows:

```

**Comment by martindurant** [comment]
Indeed, if you want a time-based window, then the index of your dataframe must be of time type - it's the values of that index which determine whether a row belongs in the window or not. An error message to this effect would have been useful! You could set the index in a separate map step.

Furthermore, summing will fail on the string column; not because you can't add strings, but because those strings will be subtracted again as rows fall out of the window, which will fail.

**Comment by inc0** [comment]
Thank you!

Do we have an example or note about time-based indices for time windows? If not, would you be able to help me draft one?
Alternatively we could add something like `inp.window(value="2s", include_timestamp_index=True)` that will automatically make it work with current timestamp.

Right, I was actually building my own aggregator for that.
